workflow/scripts/osemosys_global/transmission/main.py

```diff
@@ -0,0 +1,247 @@
+                                                               plexos_prop, interface_data,
+                                                               start_year, end_year,
+                                                               region_name)
+
+    # Adjust activity limits if cross border trade is not allowed following user config.
+    df_crossborder_final = activity_transmission_limit(cross_border_trade, df_oar_trn_final)
+    
+    # Set CapacityToActivityUnit.
+    df_capact_trn_final = capact_transmission(capact_base, df_oar_trn_final)
+    
+    # Set capital and fixed transmission costs.
+    df_cap_cost_trn_final, df_fix_cost_trn_final = get_transmission_costs(line_data, df_oar_trn_final,
+                                                                    cap_cost_base, fix_cost_base,
+                                                                    start_year, end_year,
+                                                                    region_name)
+
+    # Set operational life for transmission.
+    df_op_life_trn_final = set_op_life_transmission(df_iar_trn_final, df_oar_trn_final, 
+                                                    op_life_dict, op_life_base, region_name)
+    
+    # Set annual capacity investment constraints.
+    df_max_cap_invest_trn_final = cap_investment_constraints_trn(df_iar_trn_final, 
+                                                                 max_cap_invest_base, 
+                                                                 no_investment_techs, 
+                                                                 start_year, 
+                                                                 end_year, 
+                                                                 region_name)
+    
+    tech_set_trn_final = pd.concat([tech_set, create_tech_set_trn(interface_data)])
+
+    # Alter output csv's based on user defined capacities following user config.
+    if not tech_capacity_trn is None:
+        (tech_set_trn_final, 
+         df_max_cap_invest_trn_final, 
+         df_min_cap_invest_trn_final, 
+         df_res_cap_trn_final, 
+         df_iar_trn_final,
+         df_oar_trn_final,
+         df_op_life_trn_final, 
+         df_capact_trn_final, 
+         df_cap_cost_trn_final
+         ) = set_user_defined_capacity_trn(
+             tech_capacity_trn, 
+             op_life_dict, 
+             tech_set_trn_final, 
+             min_cap_invest_base, 
+             df_max_cap_invest_trn_final, 
+             res_cap_base,
+             df_iar_trn_final,
+             df_oar_trn_final,
+             df_op_life_trn_final,
+             df_capact_trn_final,
+             df_cap_cost_trn_final,
+             start_year,
+             end_year,
+             region_name,
+             DF_IAR_CUSTOM_VAL, 
+             DF_OAR_CUSTOM_VAL
+             )
+
+    # OUTPUT CSV's
+    
+    df_oar_trn_final.to_csv(os.path.join(output_data_dir, "OutputActivityRatio.csv"), index=None)
```

Comment:
Something is incorrect with the OAR, I believe. When we use it to build emissions, coal is giving incorrect results. 

I am using this code:

```python
df_oar = pd.read_csv(Path(output_data_dir, "OutputActivityRatio.csv"))
df_oar[(df_oar.TECHNOLOGY == "PWRCOAINDWE01") & (df_oar.YEAR == 2021)]
```

On the master branch, this is the sample datastructure. 

![image](https://github.com/user-attachments/assets/c4c3f2e7-751d-4825-b2f1-c0a683af68f9)

On this branch, this is the same output: 

![image](https://github.com/user-attachments/assets/5cf142f7-2c82-4618-8e52-f3c7afee7fed)

Reply:
I believe this is now addressed in [this commit](https://github.com/OSeMOSYS/osemosys_global/pull/206/commits/a85e95bd32bbfbed1d62b832815ad99d1ec4e01d)

Reply:
Correct! This relates to the same problem that I hinted on last week with Abhi confirming the problem originating when defining the user defined capacities. I should have made an issue as I didn't get around to a fix yet. Thanks for the commit!